### PR TITLE
Registered python meta functions to a table

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81093
* __->__ #81092

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -7,11 +7,33 @@
...
 
 from typing import List, Optional
 
+aten = torch.ops.aten
+
 meta_lib = torch.library.Library("aten", "IMPL", "Meta")
 
+meta_table = {}
+
+def register_meta(op, register_dispatcher=True):
```
Comment (anjali411) [inline torch/_meta_registrations.py:20]: What's a case when someone would want to set `register_dispatcher` to `False`?

**Review by anjali411 — APPROVED** [review]
LGTM!

File: torch/_meta_registrations.py (hunk 1 above)
Comment (Chillee) [reply torch/_meta_registrations.py:20]: If there's already a meta function registered to that operator, for example, but we want a separate one in Python.

**Pull request closed without merging**